File: Serene/Serene.Script/Northwind/Territory/TerritoryDialog.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported]
    [IdProperty("ID"), NameProperty("TerritoryID")]
    [FormKey("Northwind.Territory"), LocalTextPrefix("Northwind.Territory"), Service("Northwind/Territory")]
    public class TerritoryDialog : EntityDialog<TerritoryRow>
    {
    }
}

File: Serene/Serene.Core/Modules/Northwind/Territory/TerritoryDialog.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class TerritoryDialog extends Serenity.EntityDialog<TerritoryRow, any> {
        protected getFormKey() { return TerritoryForm.formKey; }
        protected getIdProperty() { return TerritoryRow.idProperty; }
        protected getLocalTextPrefix() { return TerritoryRow.localTextPrefix; }
        protected getNameProperty() { return TerritoryRow.nameProperty; }
        protected getService() { return TerritoryService.baseUrl; }

        protected form = new TerritoryForm(this.idPrefix);

        protected getLanguages() {
            return LanguageList.getValue();
        }
    }
}